gkr_iop/src/gkr/layer.rs

```diff
@@ -257,10 +257,60 @@ impl<E: ExtensionField> Layer<E> {
 }
 
 impl<'a, E: ExtensionField> LayerWitness<'a, E> {
-    pub fn new(wits: Vec<ArcMultilinearExtension<'a, E>>) -> Self {
-        assert!(!wits.is_empty() || !wits.is_empty());
-        let num_vars = log2(wits[0].evaluations().len()) as usize;
-        assert!(wits.iter().all(|b| b.evaluations().len() == 1 << num_vars));
-        Self { wits, num_vars }
+    pub fn new(
+        wits: Vec<ArcMultilinearExtension<'a, E>>,
+        fixed: Vec<ArcMultilinearExtension<'a, E>>,
+    ) -> Self {
+        let mut wits_and_fixed = wits;
+        wits_and_fixed.extend(fixed);
+        assert!(!wits_and_fixed.is_empty());
+        let num_vars = log2(wits_and_fixed[0].evaluations().len()) as usize;
+        assert!(
```

Comment:
nitpick: all of check seems could be simplified to one line?

```rust
assert!(
     wits_and_fixed
                .iter()
                .map(|b| b.num_vars())
                .all_equal()
)
```

Reply:
Got it! It's certainly better :smile:

Reply:
Got it! It's certainly better :smile: Fixed in https://github.com/scroll-tech/ceno/pull/970/commits/5682bd995729ad270e9a4995fbf3f6880ad9ec0b.